crates/anvil/tests/it/optimism.rs

```diff
@@ -165,52 +164,60 @@ async fn test_deposit_transaction_hash_matches_sepolia() {
         "7ef861a0dfd7ae78bf3c414cfaa77f13c0205c82eb9365e217b2daa3448c3156b69b27ac94778f2146f48179643473b82931c4cd7b8f153efd94778f2146f48179643473b82931c4cd7b8f153efd872386f26fc10000872386f26fc10000830186a08080",
     )
     .unwrap();
-    let deposit_tx_from = "0x778F2146F48179643473B82931c4CD7B8F153eFd".parse::<Address>().unwrap();
 
     let provider = http_provider_with_signer(&handle.http_endpoint(), signer.clone());
 
-    // TODO: necessary right now because transaction validation fails for deposit tx
-    //  with `from` account that doesn't have sufficient ETH balance.
-    // Should update the tx validation logic for deposit tx to
-    // 1. check if `tx.value > account.balance + tx.mint`
-    // 2. don't check `account.balance > gas * price + value` (the gas costs have been prepaid on
-    //    L1)
-    // source: https://specs.optimism.io/protocol/deposits.html#execution
-    let fund_account_tx = TransactionRequest::default()
-        .with_chain_id(31337)
-        .with_nonce(0)
-        .with_from(sender_addr)
-        .with_to(deposit_tx_from)
-        .with_value(U256::from(1e18))
-        .with_gas_limit(21_000)
-        .with_max_fee_per_gas(20_000_000_000)
-        .with_max_priority_fee_per_gas(1_000_000_000);
-
-    provider
-        .send_transaction(WithOtherFields::new(fund_account_tx))
+    let receipt = provider
+        .send_raw_transaction(raw_deposit_tx.as_slice())
         .await
         .unwrap()
-        .register()
+        .get_receipt()
         .await
         .unwrap();
 
-    // mine block
-    api.evm_mine(None).await.unwrap();
+    assert_eq!(receipt.transaction_hash, tx_hash);
+}
 
-    let pending = provider
-        .send_raw_transaction(raw_deposit_tx.as_slice())
-        .await
-        .unwrap()
-        .register()
-        .await
-        .unwrap();
+#[tokio::test(flavor = "multi_thread")]
+async fn test_deposit_tx_checks_sufficient_funds_after_applying_deposited_value() {
+    // enable the Optimism flag
+    let (_api, handle) =
+        spawn(NodeConfig::test().with_optimism(true).with_hardfork(Some(Hardfork::Paris))).await;
 
-    // mine block
-    api.evm_mine(None).await.unwrap();
```

Comment:
there was a small race condition here.

If the deposit-tx is sent then automined, it isn't evicted from the tx pool fast enough (with `Pool.prune_markers()`) so the same tx is included in the next block as well.